# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/test/gtest/TestCookie.cpp
# Commit: d01a1a055fec
# Full Hash: d01a1a055fece12b906b938954f984bd69278ef7
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-30 20:49:53
# Regressor Bug: 1960824
# File Overlap Count: 1
# Description:
#   Bug 1960824 - Introduce nsICookie::SAMESITE_UNSET, r=edgul,webdriver-reviewers,extension-reviewers,cookie-reviewers,valentin,Sasha,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246638
# ==============================================================================

diff -r 307ff391dd02 -r d01a1a055fec netwerk/test/gtest/TestCookie.cpp
--- a/netwerk/test/gtest/TestCookie.cpp	Wed Apr 30 08:53:12 2025 +0000
+++ b/netwerk/test/gtest/TestCookie.cpp	Wed Apr 30 08:59:18 2025 +0000
@@ -942,12 +942,14 @@
     int32_t sameSiteAttr;
     cookie->GetSameSite(&sameSiteAttr);
     if (name.EqualsLiteral("unset")) {
-      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_NONE);
+      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_UNSET);
     } else if (name.EqualsLiteral("unspecified")) {
-      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_NONE);
+      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_UNSET);
     } else if (name.EqualsLiteral("empty")) {
-      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_NONE);
+      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_UNSET);
     } else if (name.EqualsLiteral("bogus")) {
+      EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_UNSET);
+    } else if (name.EqualsLiteral("none")) {
       EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_NONE);
     } else if (name.EqualsLiteral("strict")) {
       EXPECT_TRUE(sameSiteAttr == nsICookie::SAMESITE_STRICT);
@@ -998,52 +1000,6 @@
   // *** speed tests
 }
 
-TEST(TestCookie, SameSiteLax)
-{
-  Preferences::SetBool("network.cookie.sameSite.laxByDefault", true);
-
-  nsresult rv;
-
-  nsCOMPtr<nsICookieService> cookieService =
-      do_GetService(kCookieServiceCID, &rv);
-  ASSERT_NS_SUCCEEDED(rv);
-
-  nsCOMPtr<nsICookieManager> cookieMgr =
-      do_GetService(NS_COOKIEMANAGER_CONTRACTID, &rv);
-  ASSERT_NS_SUCCEEDED(rv);
-
-  EXPECT_NS_SUCCEEDED(cookieMgr->RemoveAll());
-
-  SetACookie(cookieService, "http://samesite.test", "unset=yes");
-
-  nsTArray<RefPtr<nsICookie>> cookies;
-  EXPECT_NS_SUCCEEDED(cookieMgr->GetCookies(cookies));
-  EXPECT_EQ(cookies.Length(), (uint64_t)1);
-
-  Cookie* cookie = static_cast<Cookie*>(cookies[0].get());
-  EXPECT_EQ(cookie->RawSameSite(), nsICookie::SAMESITE_NONE);
-  EXPECT_EQ(cookie->SameSite(), nsICookie::SAMESITE_LAX);
-
-  Preferences::SetCString("network.cookie.sameSite.laxByDefault.disabledHosts",
-                          "foo.com,samesite.test,bar.net");
-
-  EXPECT_NS_SUCCEEDED(cookieMgr->RemoveAll());
-
-  cookies.SetLength(0);
-  EXPECT_NS_SUCCEEDED(cookieMgr->GetCookies(cookies));
-  EXPECT_EQ(cookies.Length(), (uint64_t)0);
-
-  SetACookie(cookieService, "http://samesite.test", "unset=yes");
-
-  cookies.SetLength(0);
-  EXPECT_NS_SUCCEEDED(cookieMgr->GetCookies(cookies));
-  EXPECT_EQ(cookies.Length(), (uint64_t)1);
-
-  cookie = static_cast<Cookie*>(cookies[0].get());
-  EXPECT_EQ(cookie->RawSameSite(), nsICookie::SAMESITE_NONE);
-  EXPECT_EQ(cookie->SameSite(), nsICookie::SAMESITE_LAX);
-}
-
 TEST(TestCookie, OnionSite)
 {
   Preferences::SetBool("dom.securecontext.allowlist_onions", true);